packages/medusa/src/api/admin/collections/[id]/route.ts

```diff
@@ -26,7 +26,7 @@ export const GET = async (
 }
 
 export const POST = async (
-  req: AuthenticatedMedusaRequest<AdminUpdateCollectionType>,
+  req: AuthenticatedMedusaRequest<AdminUpdateCollectionType & AdditionalData>,
```

Comment:
thought: Would it make sense to have `WithAdditionalData` and be able to do `AuthenticatedMedusaRequest<WithAdditionalData<YourOriginalType>>` That way you don't have to manage the intersection manually and additional data can be managed by us?